src/App.js

```diff
@@ -28,38 +29,104 @@ const generateSquares = () => {
 const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
-  const [squares, setSquares] = useState(generateSquares());
+  const [squares, updateSquares] = useState(generateSquares());
+  // keeping track of players
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  // find winner
+  const [winner, setWinner] = useState(null);
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const checkSquare = (id) => {
+    for (let row of squares) {
```

Comment:
Since you don't have a check to see if there is a winner here users can continue to fill in squares after the game has ended.